src/components/Article/Comment.tsx

```diff
@@ -1,25 +1,52 @@
 import { formatDistance } from 'date-fns';
+import { AxiosResponse } from 'axios';
+import { useState } from 'react';
 
-import { IAuthor } from 'interfaces';
+import { IAuthor, IEditCommentData } from 'interfaces';
+import { EditMenu } from './EditMenu';
+import { useAuth } from 'hooks/useProvideAuth';
 
 interface ICommentProps {
   createdAt: string;
   body: string;
   author: IAuthor;
+  id: number;
+  deleteComment: (url: string) => Promise<AxiosResponse | string>;
 }
 
-export function Comment({ author, createdAt, body }: ICommentProps) {
+export function Comment({ author, createdAt, body, id }: ICommentProps) {
+  const { user } = useAuth();
+  const currentUser = user?.user.username;
+  const [editMenuIsShow, setEditMenuIsShow] = useState<IEditCommentData>({
```

Comment:
the name of this variable implies that it should be a boolean (`editMenuIsShow`).